frontend/src/components/utils/TokenAmount.tsx

```diff
@@ -0,0 +1,54 @@
+import { OverlayTrigger, Tooltip } from "react-bootstrap";
+import { formatToPowerOfTen } from "../../libraries/formatting";
+import { styled } from "../../libraries/styles";
+import { AccountFungibleToken } from "../../types/common";
+
+const Offsetted = styled("span", {
+  marginLeft: "0.3em",
+});
+
+type Props = {
+  token: AccountFungibleToken;
+  sign?: -1 | 1;
+  noSymbol?: boolean;
+};
+
+const TRAILING_ZEROES_REGEX = /(.*?)0+$/g;
+
+export const TokenAmount: React.FC<Props> = (props) => {
+  const formattedAmount = formatToPowerOfTen(
+    props.token.balance,
+    props.token.decimals
+  );
+  const remainderNoZeroes = formattedAmount.remainder.replace(
+    TRAILING_ZEROES_REGEX,
+    "$1"
+  );
+  let remainder = Number("0." + remainderNoZeroes)
+    .toFixed(3)
+    .slice(1)
+    .replace(TRAILING_ZEROES_REGEX, "$1");
+  if (remainder === ".") {
+    remainder = "";
+  }
+
+  return (
+    <OverlayTrigger
+      placement={"bottom"}
+      overlay={
+        <Tooltip id="fungible-token">
+          {`${props.sign === -1 ? "-" : ""}${formattedAmount.quotient}${
+            remainderNoZeroes.length === 0 ? "" : `.${remainderNoZeroes}`
+          } ${props.token.symbol}`}
+        </Tooltip>
+      }
+    >
+      <span>
+        {props.sign === -1 ? "-" : ""}
```

Comment:
We should display `+` when it is not `-` if we display balance *change*

<img width="554" alt="image" src="https://user-images.githubusercontent.com/304265/174879741-c3413a51-a93d-4d81-ad35-62a8d43310aa.png">